adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,67 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
+#include <tuple> 
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+enum class KindOfChar {OpenBracket, ClosedBracket, NoBracket};
+
+std::pair<KindOfChar,  char> Find_KindOfChar_CorrespondingBracket(const char & c){
```

Comment:
1. Please re-consider your naming style - that doesn't match an usual snake case nor camel case
2. Pass literal types by copy

Reply:
Ok, thank you. So, something like this is ok?
std::pair<KindOfChar,  char> find_kindofchar_correspondingbracket(char c)

Reply:
Moreover, how can I choose between snake case and camel case. Is there any situation in which one is better than the other? Can I mix them into a code or should I stay consistent with only one style (e.g only camel or only snake style)?

Reply:
I'd recommend you to choose one style and follow it (if it does not go against your project coding style)
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl8-use-a-consistent-naming-style

That's a question of readability and aesthetic
Also consider https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl10-prefer-underscore_style-names